### ISSUE TITLE
aws_spot_fleet_request keeps waiting for creation even if invalid iam_fleet_role

### ISSUE THREAD

**Issue by milliburn** [terraform-providers/terraform-provider-aws#3430]
### Description

**Terraform version:** 0.11.3
**AWS provider version:** 1.9.0

The `aws_spot_fleet_request` resource appears to disregard 4xx errors returned by Amazon when attempting to create the spot fleet. In the particular case detailed below I had mistyped the `iam_fleet_role`, which invalidated the spot fleet request altogether; the resource would regardless wait for the spot fleet request to be created up to the 10m timeout mark.

### Affected Resource(s)

- aws_spot_fleet_request

Not tested with other resources.

### Terraform Configuration Files

Relevant section with irrelevant details omitted:

```
resource "aws_spot_fleet_request" "vpn" {
  spot_price = "${var.instance_spot_price_max}"
  allocation_strategy = "lowestPrice"
  wait_for_fulfillment = true
  target_capacity = 1
  replace_unhealthy_instances = true
  instance_interruption_behaviour = "stop"
  iam_fleet_role = "awn:aws:iam::${data.aws_caller_identity.aws_account.account_id}:role/aws-ec2-spot-fleet-tagging-role"

  # (launch specifications omitted)
}
```

- Note typo (`awn`, not `arn`) in `iam_fleet_role`.

### Debug Output

Sample output with `TF_LOG=DEBUG`: https://gist.github.com/milliburn/8093bc24ba00039702c6afaf2b23678b

- The actual AWS account ID has been removed from the sample.
- Note `awn:aws:iam::ACCOUNT_ID:role/aws-ec2-spot-fleet-tagging-role` in the error message: specifically the "awn" typo.
- Note Amazon's response is 400 Bad Request.

### Expected Behavior

Resource terminates with an informative error (e.g. the `Parameter: SpotFleetRequestConfig.IamFleetRole is invalid.awn:aws:iam::ACCOUNT_ID:role/aws-ec2-spot-fleet-tagging-role does not start with arn` returned by Amazon) if the first request fails with a 400 Bad Request.

### Actual Behavior

Resource waits for the spot request to be created:

```
module.ec2_vpn.aws_spot_fleet_request.vpn: Still creating... (1m10s elapsed)
```

### Steps to Reproduce

1. `terraform apply` an aws_spot_fleet_request with an invalid fleet role.

### Related

- https://github.com/terraform-providers/terraform-provider-aws/issues/3281 is another example of a bad request being ignored.

**Comment by bflad** [comment]
Plan time validation of `iam_fleet_role` as an ARN has been merged into master (via #3431) and will be released in v1.10.0, likely at the end of this week.

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
